### ISSUE TITLE
webpack.WebpackError is not a constructor when migrating from Nx v12.5.0 to Nx v12.6.0

### ISSUE THREAD

**Issue by ashwinGokhale** [nrwl/nx#6485]
Related: #6483 

## Current Behavior
After upgrading from Nx v12.5.0 to Nx v12.6.0, when I build my React app with Webpack 4 in production mode, it throws an error:
`webpack.WebpackError is not a constructor` when running the command: `yarn nx build automation --prod`. However, when building without the `--prod` flag, it works fine.

## Expected Behavior

I expect it to build. I believe this is a regression. I think what's happening is that @nrwl/web thinks I'm using Webpack 5 when really I am still using Webpack 4.

## Steps to Reproduce
1. Clone the [nx-examples repo](https://github.com/nrwl/nx-examples)
2. Run `yarn`
3. Change line 56 in `apps/cart/project.json`:
  3.1. From `"maximumWarning": "2mb",`
  3.2. To this:  `"maximumWarning": "2b",`
5. Try to build the cart app in production mode: `yarn nx build cart --prod`

This issue may not be prioritized if details are not provided to help us reproduce the issue.

### Failure Logs
```
$ nx build cart --prod

> nx run cart:build:production
webpack.WebpackError is not a constructor

———————————————————————————————————————————————

>  NX   ERROR  Running target "cart:build" failed

  Failed tasks:

  - cart:build:production

  Hint: run the command with --verbose for more details.
```

### Environment
```
  Node : 14.16.1
  OS   : win32 x64
  yarn : 1.22.5

  nx : Not Found
  @nrwl/angular : 12.6.4
  @nrwl/cli : 12.6.4
  @nrwl/cypress : 12.6.4
  @nrwl/devkit : 12.6.4
  @nrwl/eslint-plugin-nx : 12.6.4
  @nrwl/express : Not Found
  @nrwl/jest : 12.6.4
  @nrwl/linter : 12.6.4
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/nx-cloud : 12.3.5
  @nrwl/react : 12.6.4
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.6.4
  @nrwl/web : 12.6.4
  @nrwl/workspace : 12.6.4
  @nrwl/storybook : 12.6.4
  @nrwl/gatsby : Not Found
  typescript : 4.3.5
```

**Comment by ashwinGokhale** [comment]
@jaysoo I think you need to add WebpackError to the exports in this file to maintain webpack 4/5 compatibility: https://github.com/nrwl/nx/blob/master/packages/web/src/webpack/bundle4.ts

**Comment by akryvasheyeu** [comment]
same problem after migration to 12.6.2

**Comment by legovadim** [comment]
same issue 12.6.2

**Comment by legovadim** [comment]
So quick fix: in workspace.json, build target -> production -> budgets -> Increase the maximum warning and error values.

**Comment by akryvasheyeu** [comment]
nice breaking change https://github.com/nrwl/nx/commit/369da3ee788a8ca15eee26541ed5bd4f781c2147#diff-6ec6838bce1fbd7a097ad4cae81fd5c8ba7709877e693b7286cef6e043125556 . this change is not compatible with webpack 4

**Comment by vetik32** [comment]
works in 12.5.9

**Comment by TassiloSinghammerVolue** [comment]
It also works in a fresh workspace. So it seems to be the combination of the above mentioned change and some existing configuration.

**Comment by kazamov** [comment]
@TassiloSinghammerVolue, in the version 12.6.3 the BundleBudget plugin which cause an issue disabled for Webpack 5 as a temp solution I think. That's why it is working

**Comment by ashwinGokhale** [comment]
I created a PR to fix it #6519. I'm hoping it can get reviewed soon.
